# snapshot or record
int8 type
---
int32 count